src/pybind/mgr/smb/fs.py

```diff
@@ -106,3 +121,91 @@ def resolve_exists(
                 raise NotADirectoryError(volpath)
         log.debug('Verified that %r exists in %r', volpath, volume)
         return volpath
+
+
+class _TTLCache:
+    def __init__(self, maxsize: int = 512, ttl: float = 300.0) -> None:
+        self.cache: Dict[
+            Tuple[str, str, str], Tuple[Tuple[str, str, str], float]
+        ] = {}
+        self.maxsize: int = maxsize
+        self.ttl: float = ttl
+
+    def _evict(self) -> None:
+        current_time = time.monotonic()
+
+        # Remove expired entries
+        keys_to_evict = [
+            key
+            for key, (_, timestamp) in self.cache.items()
+            if current_time - timestamp > self.ttl
+        ]
+        for key in keys_to_evict:
+            del self.cache[key]
+
+        # Ensure cache does not exceed maxsize
+        if len(self.cache) > self.maxsize:
+            sorted_keys = sorted(
+                self.cache.keys(), key=lambda k: self.cache[k][1]
+            )
+            for key in sorted_keys[: len(self.cache) - self.maxsize]:
+                del self.cache[key]
+
+    def get(
+        self, key: Tuple[str, str, str]
+    ) -> Optional[Tuple[str, str, str]]:
+        self._evict()
+        if key in self.cache:
+            value, timestamp = self.cache[key]
+            if time.monotonic() - timestamp < self.ttl:
```

Comment:
This all seems redundant with the work done in _evict. Let's drop this extra comparsion and just rely on `_evict` working as desgined.

Reply:
Sure, got rid of extra check & let `_evict` do tht job.